loadimpact/k6-docs
# Breaking change: mdx processing update

**Describe the changes**
This PR brings a long-awaited update to the way we process mdx: `Code` and `CodeGroup` now being rendered via native mdx renderer as the opposite to previously hot replacement which caused performance issues and was blocking some features to come out.

**Verbose**
- migrated from `gatsby-remark-prismjs` to `prismjs-react-renderer` for max flexibility during custom code block render
- refactored `Code` and `CodeGroup` components
- added postponed earlier collapse/expand mechanism for `Code` component ([task in Notion](https://www.notion.so/pixelpoint/Add-scrolling-option-to-code-blocks-b9f7e7d073e04e8ea5e72152ad9160cd))
- added `CodeInline` component
- updated code blocks usage in every `.md` file

**Side**
- Fixed `Blockquote` component styling bugs described [here](https://github.com/loadimpact/k6-docs/commit/b657e01b7ca3980337571682c839f15a35814008#commitcomment-43306583)
- Since `Cloud API` section has been published, I've enabled Algolia indexing for them back 

**Steps to test**
1. Skim check the diff in `md` files, make sure nothing important got deleted accidentally
2. Make sure it works locally both in dev and prod 
3. Make some random page with code blocks checks, see everything is working as expected
4. Find any page with large code blocks, test expand/collapse mechanism (or just copy past content of a small one temporarily)
